core/error_macros.h

```diff
@@ -173,6 +193,18 @@ extern bool _err_error_exists;
 			_err_error_exists = false;                                                                     \
 	}
 
+/** Use this one if there is no sensible fallback, that is, the error is unrecoverable.
+ */
+
+#define ERR_FAIL_COND_TRAP(m_cond)                                                                         \
+	{                                                                                                      \
+		if (m_cond) {                                                                                      \
+			_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Condition ' " _STR(m_cond) " ' is true."); \
```

Comment:
I think you might like to add a `"Fatal: "` before `"Condition...`, since otherwise this would look the same as a `ERR_FAIL_COND` followed by a crash, while it is actually a meant-to-be crash.